packages/discord.js/src/structures/MessagePayload.js

```diff
@@ -248,7 +248,13 @@ class MessagePayload {
       components,
       username,
       avatar_url: avatarURL,
-      allowed_mentions: content === undefined && message_reference === undefined ? undefined : allowedMentions,
+      allowed_mentions:
+        content === undefined &&
+        message_reference === undefined &&
+        // Components V2 can mention, so need to check for them too
+        (!components?.length || flags & (MessageFlags.IsComponentsV2 === 0))
```

Comment:
```suggestion
        (!components?.length || (flags & MessageFlags.IsComponentsV2) === 0)
```

Reply:
Did a complete refactor instead, as the original intention of this was to prevent passing allowedMentions if editing another user's message's flags.